spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ReactorSleuth.java

```diff
@@ -55,15 +54,20 @@ private ReactorSleuth() {
 	 * {@link reactor.core.publisher.Hooks#onLastOperator(Function)} or
 	 * {@link reactor.core.publisher.Hooks#onLastOperator(Function)}. The Span operator
 	 * pointcut will pass the Scope of the Span without ever creating any new spans.
-	 * @param beanFactory - {@link BeanFactory}
+	 * @param springContext the Spring context.
 	 * @param <T> an arbitrary type that is left unchanged by the span operator
 	 * @return a new lazy span operator pointcut
 	 */
-	@SuppressWarnings("unchecked")
+	// Much of Boot assumes that the Spring context will be a
+	// ConfigurableApplicationContext, rooted in SpringApplication's
+	// requirement for it to be so. Previous versions of Reactor
+	// instrumentation injected both BeanFactory and also
+	// ConfigurableApplicationContext. This chooses the more narrow
+	// signature as it is simpler than explaining instanceof checks.
 	public static <T> Function<? super Publisher<T>, ? extends Publisher<T>> scopePassingSpanOperator(
```

Comment:
That's a breaking change. Maybe we should deprecate this one and create a new one?

Reply:
I looked on github, nothing is using it except us.. deprecating would cause a lot of debt

Reply:
ex https://github.com/search?q=reactorsleuth&type=Code only copies of sleuth

Reply:
:+1: